compute/main.go

```diff
@@ -0,0 +1,656 @@
+					},
+					//DiskSizeGB: to.Int32Ptr(100), // default 127G
+				},
+			},
+			HardwareProfile: &armcompute.HardwareProfile{
+				VMSize: armcompute.VirtualMachineSizeTypesStandardF2S.ToPtr(), // VM size include vCPUs,RAM,Data Disks,Temp storage.
+			},
+			OSProfile: &armcompute.OSProfile{ //
+				ComputerName:  to.StringPtr("sample-compute"),
+				AdminUsername: to.StringPtr("sample-user"),
+				AdminPassword: to.StringPtr("Password01!@#"),
+				//require ssh key for authentication on linux
+				//LinuxConfiguration: &armcompute.LinuxConfiguration{
+				//	DisablePasswordAuthentication: to.BoolPtr(true),
+				//	SSH: &armcompute.SSHConfiguration{
+				//		PublicKeys: []*armcompute.SSHPublicKey{
+				//			{
+				//				Path:    to.StringPtr(fmt.Sprintf("/home/%s/.ssh/authorized_keys", "sample-user")),
+				//				KeyData: to.StringPtr(string(sshBytes)),
+				//			},
+				//		},
+				//	},
+				//},
+			},
+			NetworkProfile: &armcompute.NetworkProfile{
+				NetworkInterfaces: []*armcompute.NetworkInterfaceReference{
+					{
+						SubResource: armcompute.SubResource{
+							ID: to.StringPtr(networkInterfaceID),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	pollerResponse, err := vmClient.BeginCreateOrUpdate(ctx, resourceGroupName, vmName, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.VirtualMachine, nil
+}
+
+func deleteVirtualMachine(ctx context.Context, connection *armcore.Connection) (*http.Response, error) {
+	vmClient := armcompute.NewVirtualMachinesClient(connection, subscriptionId)
+
+	pollerResponse, err := vmClient.BeginDelete(ctx, resourceGroupName, vmName, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func createDisk(ctx context.Context, connection *armcore.Connection, diskName string, diskSize int32) (*armcompute.Disk, error) {
+	diskClient := armcompute.NewDisksClient(connection, subscriptionId)
+
+	disk := armcompute.Disk{
+		Resource: armcompute.Resource{
+			Location: to.StringPtr(location),
+		},
+		Properties: &armcompute.DiskProperties{
+			CreationData: &armcompute.CreationData{
+				CreateOption: armcompute.DiskCreateOptionEmpty.ToPtr(), // create
+			},
+			DiskSizeGB: to.Int32Ptr(diskSize),
+		},
+	}
+
+	pollerResponse, err := diskClient.BeginCreateOrUpdate(ctx, resourceGroupName, diskName, disk, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+	return resp.Disk, nil
+}
+
+func deleteDisk(ctx context.Context, connection *armcore.Connection, diskName string) (*http.Response, error) {
+	diskClient := armcompute.NewDisksClient(connection, subscriptionId)
+
```

Comment:
Looks like these two functions are not used in this sample. We could remove these and use them in some other samples for more complicated scenarios - for instance attach disks to an existing virtual machine.
Therefore could we remove these two functions just for this scenario?